### PR TITLE
Upgrade parser

### PR DESCRIPTION
**Pull request by rplzzz**
Parse configuration file using ConfigObj, instead of kludgey homebrew (feat. regular expressions!).  Also, remove the `eval` kludge.  There is now a factory function that will create component instances given their name, or raise an exception if the component type is unknown.

I had intended to separate the parsing and the bootstrapping of the running system, but at this point the parsing is pretty trivial, so the bootstrapping code calls the parser directly and then proceeds to do its thing.

### CODE REVIEW

File: cassandra/cassandra_main.py [hunk 1]
```diff
@@ -15,83 +15,40 @@
 import threading
 import argparse
 
+def bootstrap_sp(cfgfile_name):
```
Comment (calebbraun) [inline cassandra/cassandra_main.py]: What is `sp`?

Comment (rplzzz) [reply cassandra/cassandra_main.py]: "Single processing".  As contrasted with `mp` for multiprocessing.  I'm open to other names if you don't like that one.  There are, as they say, only two hard things in computer science:  cache invalidation, and naming things.

Comment (calebbraun) [reply cassandra/cassandra_main.py]: The name is fine with me as long as the words "single processing" appear somewhere near the declaration or function call.

File: example.cfg [hunk 2]
```diff
@@ -3,24 +3,25 @@ ModelInterface = /people/link593/wrk/ModelInterface-baseX/ModelInterface.jar
 DBXMLlib = /people/link593/lib
 inputdir = ./input-data
 rgnconfig = rgn32
+modtable = module-table.csv
```
Comment (calebbraun) [inline example.cfg]: No longer our approach.

File: module-table.csv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Module Name,Config Section
```
Comment (calebbraun) [inline module-table.csv]: Don't need.

File: cassandra/cassandra_main.py (hunk 1 above)
Comment (rplzzz) [reply cassandra/cassandra_main.py]: Sounds reasonable.  Could you just add it wherever you feel it fits best?  I've got another branch checked out right now, and I'm juggling a bunch of WIP.

File: example.cfg (hunk 2 above)
Comment (rplzzz) [reply example.cfg]: Thanks, I'd missed this one and the one below.  You should be able to delete these via GitHub's new suggestion feature.

**Pull request merged**